modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm

```diff
@@ -771,22 +819,25 @@
 	if(!.)
 		return
 
-
 	var/obj/item/coomer = new /obj/item/hand_item/coom(user)
 	var/mob/living/carbon/human/target = user
 	var/obj/item/held = user.get_active_held_item()
 	var/obj/item/unheld = user.get_inactive_held_item()
-	if(user.put_in_hands(coomer) && target.dna.species.mutant_bodyparts["testicles"] && target.dna.species.mutant_bodyparts["penis"])
-		if(held || unheld)
-			if(!((held.name == "cum" && held.item_flags == DROPDEL | ABSTRACT | HAND_ITEM) || (unheld.name == "cum" && unheld.item_flags == DROPDEL | ABSTRACT | HAND_ITEM)))
-				to_chat(user, span_notice("You mentally prepare yourself to masturbate."))
-			else
-				qdel(coomer)
-		else
-			to_chat(user, span_notice("You mentally prepare yourself to masturbate."))
-	else
+
+	if(!(user.put_in_hands(coomer)) || !(target.dna.species.mutant_bodyparts["testicles"]) || !(target.dna.species.mutant_bodyparts["penis"]))
 		qdel(coomer)
 		to_chat(user, span_warning("You're incapable of masturbating."))
+		return FALSE
+
+	if(held || unheld)
+		if(!((held.name == "cum" && held.item_flags == DROPDEL | ABSTRACT | HAND_ITEM) || (unheld.name == "cum" && unheld.item_flags == DROPDEL | ABSTRACT | HAND_ITEM)))
```

Comment:
we should never ever be checking an item's name to determine behavior

Reply:
sex2 code in a nutshell lmao